packages/calcite-components/src/utils/interactive.tsx

```diff
@@ -153,7 +157,7 @@ function removeInteractionListeners(element: HTMLElement): void {
  * @param component
  */
 export function connectInteractive(component: InteractiveComponent): void {
-  if (!component.disabled || !isFirefox) {
```

Comment:
If the component doesn't start out disabled, the parent is never stored so the later part of the workaround can't work.